linux/LinuxProcessList.c

```diff
@@ -1640,8 +1684,13 @@ static bool LinuxProcessList_recurseProcTree(LinuxProcessList* this, openat_arg_
       }
       #endif
 
-      if (ss->flags & PROCESS_FLAG_LINUX_CGROUP) {
-         LinuxProcessList_readCGroupFile(lp, procFd);
+      /* The process' cgroup name is required for checking whether it runs in a container */
+      LinuxProcessList_readCGroupFile(lp, procFd);
```

Comment:
Instead of reading the cgroup unconditional, make this check to succeed on the flag being selected OR the container setting active. You could then further put the below code as part of the block reading/processing the cgroup value.